app/main.py

```diff
@@ -32,13 +35,27 @@ def main() -> None:
     ]
 
     # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    await run_sequence(
+        run_parallel(
+            service.send_msg(wake_up_program[0]),
+            service.send_msg(wake_up_program[1])
+        ),
+        service.send_msg(wake_up_program[2])
```

Comment:
The `send_msg` method is expected to be asynchronous, so you should await each call to `service.send_msg` within `run_parallel` and `run_sequence`. Ensure that all calls to asynchronous functions are awaited to prevent any unexpected behavior.